logger/include/main.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <iostream>
+#include <source_location>
+
+/*
+ * @brief Define API's to append log messages to a buffer
```

Comment:
add doxygen just before the function declaration. and keep namespace in outer most.

Reply:
Maintain this formate for doxygen-

* @brief This API does ........in 1 line
*  more description if needed
* 
* @ param[in] message -- tell its purpose 
*  @ param[in] locationDetails -- tell its purpose 
*  @ param[out] out -- in case you have to share some output to called function.
*
* @return -- what It returns in good case/ bad case , if void keep it blank.


Something can be changed also based on need, but in general we use this format.